app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/PlantCreationViewModel.kt

```diff
@@ -31,12 +35,19 @@ class PlantCreationViewModel(
                 Plant(
                     Plant.Name(plantName),
                     speciesName,
-                    "",
+                    selectedPicture.value.orEmpty(),
                     wateringFrequency.toInt()
                 )
             )
 
             _toNavigateBack.value = Event(Unit)
         }
     }
+
+    fun onImageClicked(activityResultRegistry: ActivityResultRegistry) {
+        val imagePicker = ImagePicker(activityResultRegistry) {
+            _selectedPicture.value = it?.toString()
+        }
+        imagePicker.pickImage()
+    }
```

Comment:
I think that we can just register callback on fragment, and on viewmodel just set Uri , instead of starting this from viewmodel